actions/restore-snapshot

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# THIS ACTION RELIES HEAVILLY ON THE EXISTENCE OF A RESOURCE
+
+# Stop Jenkins
+service jenkins stop
+
+RESOURCE_FILE=$(resource-get workspace)
+RESOURCE_FINGERPRINT=$(sha256sum $RESOURCE_FILE)
+ATOMIC_OPERATION=$(action-get atomic)
+
+if [ ${ATOMIC_OPERATION} != 'False' ]
+then
+    rm -rf /var/lib/jenkins/jobs/*
+    action-set workspace.cleaned="True"
+fi
+
+action-set workspace.sum="$RESOURCE_FINGERPRINT"
+
+# Remove any stale files that may be here
+if [ -f '/var/lib/jenkins/jobs/workspaces.tgz']; then
```

Comment:
I see we are doing a pre-delete here, but I think a post delete will keep our snapshot action cleaner.